### PR TITLE
feat(sca): dockerfile image-referencer fixes

### PR DESCRIPTION
**Pull request by ChanochShayner**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description
 - Fix relatedResourceId in dockerfile IR.
 - Use `is_docker_file` in persist repo.
 - Add `result_instruction` to the resource.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ayajbara — APPROVED** [review]
🥇

File: checkov/dockerfile/checks/ReferenceLatestTag.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def scan_resource_conf(self, conf: list[_Instruction]) -> tuple[CheckResult, lis
                 return CheckResult.FAILED, [content]
             elif base_image.endswith(":latest"):
                 return CheckResult.FAILED, [content]
-        return CheckResult.PASSED, None
+        return CheckResult.PASSED, [content]
```
Comment (gruebel) [inline checkov/dockerfile/checks/ReferenceLatestTag.py:37]: nice 💯

File: tests/dockerfile/image_referencer/test_manager.py [hunk 2]
```diff
@@ -37,6 +37,6 @@ def test_extract_images_from_resources():
             name="php:7.1-apache",
             start_line=1,
             end_line=1,
-            related_resource_id="/Dockerfile.FROM",
+            related_resource_id="/Dockerfile:/Dockerfile.FROM",
```
Comment (gruebel) [inline tests/dockerfile/image_referencer/test_manager.py:40]: is it correct with the `/` in front of `Dockerfile.FROM`?

**Review by gruebel — APPROVED** [review]
🏆

File: tests/dockerfile/image_referencer/test_manager.py (hunk 2 above)
Comment (ChanochShayner) [reply tests/dockerfile/image_referencer/test_manager.py:40]: Yes

**Pull request merged**